### PR TITLE
Update Socket.io & add the automated first tests, registering and logging in.

### PR DESCRIPTION
**Pull request by kenerwin88**
## Description
This PR makes a breaking change, by updating Socket.io from v1.2.0 to v4.4.1 (latest).  The newer versions of Socket.io are much more efficient, more easily testable, and just all around better.  However, for clients to connect, they must use an updated socket.io.js file (located under js/libs/socket.io.js).  The updated version can be found here: 

https://raw.githubusercontent.com/socketio/socket.io-client/master/dist/socket.io.min.js

(This should replace THIS file https://github.com/Andaroth/MMORPGMaker-MZ/blob/master/js/libs/socket.io.js)

## Motivation and Context
Currently we have no automated testing.  The first thing I wanted to add is a basic "register" & "login" flow, that way we can verify after refactoring that at the very least, someone can login successfully.  It's VERY difficult to do that with the old socket.io version.  I've added a sample test as well to the PR, that will now run automatically on each PR.  (Yes it's a tiny test, but it's a good start!)

## How Has This Been Tested?
Tested manually, as well as via automated test.  It spins up a sqlite database in-memory, so each test is clean.

## Screenshots (if appropriate):
N/A

### CODE REVIEW

File: src/modules/player/player.ts [hunk 1]
```diff
@@ -127,6 +127,25 @@ export class Player {
                 });
             });
 
+            client.on('player_update_busy', function (payload) {
```
Comment (Mouradif) [inline src/modules/player/player.ts:130]: the event was updated to `player_update_status1

File: src/modules/player/player.ts [hunk 2]
```diff
@@ -127,6 +127,25 @@ export class Player {
...
+                if (client.playerData === undefined) {
+                    return;
+                }
+                if (client.playerData.isBusy === payload) {
```
Comment (Mouradif) [inline src/modules/player/player.ts:134]: the property was updated to `playerData.status`

File: src/modules/player/player.ts [hunk 3]
```diff
@@ -127,6 +127,25 @@ export class Player {
...
+                    return;
+                }
+
+                client.playerData.isBusy = payload;
```
Comment (Mouradif) [inline src/modules/player/player.ts:138]: 👆

File: src/modules/player/player.ts [hunk 4]
```diff
@@ -127,6 +127,25 @@ export class Player {
...
+
+                database
+                    .savePlayer({ username: client.playerData.username, isBusy: client.playerData.isBusy })
```
Comment (Mouradif) [inline src/modules/player/player.ts:141]: 👆

**Review by Mouradif — COMMENTED** [review]
The `isBusy` is a blocking issue, I'm surprised it worked on your end, if you have the latest client (on `develop`), it shouldn't.

The rest is not really blocking but I'd love to read some replies

File: package.json [hunk 5]
```diff
@@ -1,58 +1,73 @@
...
+        "preset": "ts-jest",
+        "transform": {
+            "^.+\\.(ts|tsx)?$": "ts-jest",
+            "^.+\\.(js|jsx)$": "babel-jest"
+        },
+        "testPathIgnorePatterns": ["dist"]
+    },
     "engines": {
-        "node": ">=0.12.0"
+        "node": "v17.x"
```
[the remark this answers is not in the document]
Comment (kenerwin88) [reply package.json]: Fixed, set a v16.x

File: src/mmo.ts [hunk 6]
```diff
@@ -1,22 +1,60 @@
...
 import express from 'express';
 import bodyParser from 'body-parser';
 import path from 'path';
+const cors = require('cors');
 
 const app = express();
+let Sentry;
+// Use Sentry for Telemetry if enabled
+if (process.env.USE_SENTRY === 'true') {
+    const express = require('express');
```
[the remark this answers is not in the document]
Comment (kenerwin88) [reply src/mmo.ts]: Removed, ty!

File: src/modules/player/player.ts (hunk 2 above)
Comment (kenerwin88) [reply src/modules/player/player.ts:134]: Lies!

**Review by Mouradif — APPROVED** [review]
I must not tell lies

**Pull request merged**